### PR TITLE
clip function for primitive types added

### PR DESCRIPTION
**Pull request by cagriyildirimR**
There was a small typo in README

Clip operation is missing for Complex types because they are not Comparable. I am also not sure how to implement it. Should we compare them based on their real part only?

If Comparable interface were added to complex arrays then I think this function should work without any refactoring.

If I am missing anything please let me know.

### CODE REVIEW

**Comment by devcrocod** [comment]
Hi, Cargi!
Thanks for the pull request!

Complex types aren't comparable in the mathematical sense. We deliberately did not come up with our own comparison methods to avoid UB. At the same time, in some methods, you can pass your own comparator, so you can explicitly determine how to compare complex numbers.

**Comment by devcrocod** [comment]
Thanks for the fix in the readme.
With the new release, we will completely rewrite it, since the library will change a lot.

**Comment by devcrocod** [comment]
@zaleslaw 
I think it will be interesting for you to take a look at this pr

File: multik-api/src/commonMain/kotlin/org/jetbrains/kotlinx/multik/ndarray/operations/Transformation.kt [hunk 1]
```diff
@@ -265,3 +265,27 @@ internal fun <T, D : Dimension, O : Dimension> concatenate(
...
+ * values smaller than [min] are set to [min]
+ *
+ * @param min minimum value for clipping where any value in ndarray
+ * that is smaller than [min] are clipped and set to [min]
+ * @param max maximum value for clipping where any value in ndarray
+ * that is bigger than [max] are clipped and set to [max]
+ * @return NDArray of which all of its elements are in range min..max
+ * @throws IllegalArgumentException if min > max
+ */
+internal fun <T: Comparable<T>, D: Dimension> MultiArray<T, D>.clip(min: T, max: T): NDArray<T,D> {
```
Comment (devcrocod) [inline multik-api/src/commonMain/kotlin/org/jetbrains/kotlinx/multik/ndarray/operations/Transformation.kt:282]: can be more specific type T:
``` ...): NDArray<T, D> where T : Number, T: Comparable<T>```

File: multik-api/src/commonMain/kotlin/org/jetbrains/kotlinx/multik/ndarray/operations/Transformation.kt [hunk 2]
```diff
@@ -265,3 +265,27 @@ internal fun <T, D : Dimension, O : Dimension> concatenate(
...
+    require(min <= max) {
+        "min value for clipping should be lower than or equal to the [max] value"
+    }
+    val clippedData = initMemoryView(size, dtype) { index ->
+        val e = data[index]
+        if (e < min) min else if (e > max) max else e
+    }
+    return NDArray(data = clippedData, shape = shape, dim = dim)
```
Comment (devcrocod) [inline multik-api/src/commonMain/kotlin/org/jetbrains/kotlinx/multik/ndarray/operations/Transformation.kt:290]: Don't forget that primitives arrays are passed by reference. So shape is passed here by reference, so changing the shape in the first ndarray will change the shape in the resulting ndarray from the clip function.

**Review by devcrocod — CHANGES_REQUESTED** [review]

**Comment by zaleslaw** [comment]
yeah @devcrocod it looks like a missed chain!

**Review by devcrocod — APPROVED** [review]

**Pull request merged**